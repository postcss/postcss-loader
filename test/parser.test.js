import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from './helpers';

describe('"parser" option', () => {
  it('should work with "String" value', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        parser: 'sugarss',
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.sss', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with "Object" value', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        // eslint-disable-next-line global-require,import/no-dynamic-require
        parser: require('sugarss'),
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.sss', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with "Function" value', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        // eslint-disable-next-line global-require,import/no-dynamic-require
        parser: require('sugarss').parse,
      },
    });
    const stats = await compile(compiler);

    const codeFromBundle = getCodeFromBundle('style.sss', stats);

    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should throw an error on "unresolved" parser', async () => {
    const compiler = getCompiler('./sss/index.js', {
      postcssOptions: {
        parser: 'unresolved',
      },
    });

    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats, true)).toMatchSnapshot('errors');
  });
});
