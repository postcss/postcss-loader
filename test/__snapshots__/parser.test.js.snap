// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"parser" option should throw an error on "unresolved" parser: errors 1`] = `
Array [
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
TypeError: parser is not a function",
  "ModuleError: Module Error (from \`replaced original path\`):
Loading PostCSS \\"unresolved\\" parser failed: Cannot find module 'unresolved' from 'src/utils.js'",
]
`;

exports[`"parser" option should throw an error on "unresolved" parser: warnings 1`] = `Array []`;

exports[`"parser" option should work with "Function" value: css 1`] = `
"a {
  color: black
}
"
`;

exports[`"parser" option should work with "Function" value: errors 1`] = `Array []`;

exports[`"parser" option should work with "Function" value: warnings 1`] = `Array []`;

exports[`"parser" option should work with "Object" value: css 1`] = `
"a {
  color: black
}
"
`;

exports[`"parser" option should work with "Object" value: errors 1`] = `Array []`;

exports[`"parser" option should work with "Object" value: warnings 1`] = `Array []`;

exports[`"parser" option should work with "String" value: css 1`] = `
"a {
  color: black
}
"
`;

exports[`"parser" option should work with "String" value: errors 1`] = `Array []`;

exports[`"parser" option should work with "String" value: warnings 1`] = `Array []`;
